### PR TITLE
Metrics advisor aad

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py [hunk 1]
```diff
@@ -153,7 +154,8 @@ class MetricsAdvisorAdministrationClient(object):  # pylint:disable=too-many-pub
     :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
         for example: https://:code:`<resource-name>`.cognitiveservices.azure.com).
     :param credential: An instance of ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential.
-        Requires both subscription key and API key.
+        which requires both subscription key and API key. Or an object which can provide an access
+        token for the vault, such as a credential from :mod:`azure.identity`
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py:158]: maybe this?

```suggestion
        token for the Metrics Advisor service, such as a credential from :mod:`azure.identity`
```

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py [hunk 2]
```diff
@@ -153,7 +154,8 @@ class MetricsAdvisorAdministrationClient(object):  # pylint:disable=too-many-pub
...
     :type credential: ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py:159]:
```suggestion
    :type credential: ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential or ~azure.core.credentials.TokenCredential
```

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py [hunk 3]
```diff
@@ -179,12 +181,26 @@ def __init__(self, endpoint, credential, **kwargs):
...
-        )
+        if isinstance(credential, MetricsAdvisorKeyCredential):
+            self._client = _Client(
+                endpoint=endpoint,
+                sdk_moniker=SDK_MONIKER,
+                authentication_policy=MetricsAdvisorKeyCredentialPolicy(credential),
+                **kwargs
+            )
+        else:
+            scope = "https://cognitiveservices.azure.com/.default"
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py:192]: We shouldn't hardcode the scopes, if a user is using a different cloud the [value will be slightly different](https://github.com/Azure/azure-sdk-for-net/issues/17192) so we need to allow for a custom value to be passed. 

The generated code allows for this, though, so I think we just need to regenerate with the `--add-credentials` flag which will give us [modifiable credential scopes](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_generated/_configuration.py#L46) and default to the `BearerTokenCredentialPolicy` so we don't have to add it.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py [hunk 4]
```diff
@@ -54,6 +54,28 @@ async def authentication_administration_client_with_metrics_advisor_credential_a
     # [END administration_client_with_metrics_advisor_credential_async]
 
 
+async def authentication_client_with_aad_async():
+    # [START authentication_client_with_aad_async]
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py:58]: Can we add a comment here about setting the env vars for DefaultAzureCredential? Something like: 

"DefaultAzureCredential will use the values from these environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET"

File: sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py [hunk 5]
```diff
@@ -54,6 +54,28 @@ async def authentication_administration_client_with_metrics_advisor_credential_a
...
+
+    service_endpoint = os.getenv("METRICS_ADVISOR_ENDPOINT")
+    credential = DefaultAzureCredential()
+    client = MetricsAdvisorAdministrationClient(service_endpoint, credential)
+    # [END authentication_administration_client_with_aad_async]
+
+
 async def main():
     await authentication_client_with_metrics_advisor_credential_async()
     await authentication_administration_client_with_metrics_advisor_credential_async()
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py:81]: aad samples missing in main

File: sdk/metricsadvisor/azure-ai-metricsadvisor/tests/test_metrics_advisor_client_live.py [hunk 6]
```diff
@@ -60,7 +60,7 @@ def test_list_incident_root_cause(self):
             detection_configuration_id=self.anomaly_detection_configuration_id,
             incident_id=self.incident_id,
         ))
-        assert len(results) > 0
+        assert len(results) == 0
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/tests/test_metrics_advisor_client_live.py:63]: we should see at least 1 incident root cause here using 
detection_config_id = "fb5a6ed6-2b9e-4b72-8b0c-0046ead1c15c"
incident_id = "2583ff47fef9174e6d6bfe73415ad914-174d2024c00"

(those are the values the live tests use)

File: sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py (hunk 4 above)
Comment (xiangyan99) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/samples/async_samples/sample_authentication_async.py:58]: DefaultAzureCredential is a chain of credentials and does not require AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET to work. e.g. if users have managed identity configured, they don't need to configure anything to make DefaultAzureCredential work.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py (hunk 3 above)
Comment (xiangyan99) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py]: There is an open PR for the swagger https://github.com/Azure/azure-rest-api-specs/pull/10862. Let's regenerate the code when it is settled.

**Comment by KarishmaGhiya** [comment]
@xiangyan99 i see 368 files changed in this PR - did the rebase go bad?

**Comment by xiangyan99** [comment]
> @xiangyan99 i see 368 files changed in this PR - did the rebase go bad?

No. This is by design. We have hundreds of tests. :)

You can ignore all files ending with .yaml. They are test recordings.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py [hunk 7]
```diff
@@ -59,7 +60,8 @@ class MetricsAdvisorClient(object):
     :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
         for example: https://:code:`<resource-name>`.cognitiveservices.azure.com).
     :param credential: An instance of ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential.
-        Requires both subscription key and API key.
+        which requires both subscription key and API key. Or an object which can provide an access
+        token for the vault, such as a credential from :mod:`azure.identity`
```
Comment (kristapratico) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py:64]: Can we update the credential docstring for the other clients as well?

**Review by kristapratico — APPROVED** [review]

File: sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py (hunk 7 above)
Comment (xiangyan99) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py:64]: Good catch! 👍

**Pull request merged**